### PR TITLE
Eviction admitter: Use KubeVirt generated clientset

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
Before this PR:
PodEvictionAdmitter:
1. Had exported (public) fields and could be instantiated incorrectly.
2. Used the old `kubecli.KubevirtClient` clientset.
3. Used gomock in unit tests.

After this PR:
PodEvictionAdmitter:
1. Has a factory function and only unexported (private) fields.
2. Uses kubernetes' official generated clientset.
3. Uses KubeVirt's generated clientset.
4. Uses KubeVirt's generated fake clientset in unit tests instead of gomock.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer
This is a follow up to https://github.com/kubevirt/kubevirt/pull/11380.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [X] PR: The PR description is expressive enough and will help future contributors
- [X] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by orelmisan** [comment]
[Change 1](https://github.com/kubevirt/kubevirt/compare/5cebaa4cdeb19b8409d2d9ea6328bceb036153ee..d80218df7657711ef0f7dbb2496f19be4be0cf52): Remove the permission of gomock to return the fake k8s clientset.
[Change 2](https://github.com/kubevirt/kubevirt/compare/d80218df7657711ef0f7dbb2496f19be4be0cf52..24839d4f51e64908615ee4a7a2b15330aa63853d): Fix linter issue.

**Review by enp0s3 — COMMENTED** [review]
@orelmisan Thank you!
/approve

**Review by fossedihelm — COMMENTED** [review]
Thank you @orelmisan!
Even if the generated client(`kubevirt.Interface`) provides the same interface of our custom KubevirtClient `kubecli.KubevirtClient` we need to keep in mind that it currently does not implement all the custom (`_expansion` file) methods such as `Pause`, `Freeze`, etc...
A step ahead in this direction is provided by https://github.com/kubevirt/kubevirt/pull/11928, but it still does not implement some methods that require `clientConfig`.
Here we only use `VirtualMachineInstances.Get()` so everything works.
But we must be aware that the generated client that we are using with this patch still does not cover all the methods provided by the custom KubevirtClient.
@enp0s3 FYI

**Comment by enp0s3** [comment]
@fossedihelm Thanks for the heads up! shouldn't the test fail in case the code will try to use the custom API calls that aren't implemented by the current client set?

File: pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_test.go [hunk 1]
```diff
@@ -541,6 +486,26 @@ func newDeniedAdmissionResponse(message string) *admissionv1.AdmissionResponse {
 	}
 }
 
+func newExpectedPatch(vmi *virtv1.VirtualMachineInstance) testing.PatchActionImpl {
```
Comment (RamLavi) [inline pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_test.go:489]: the name newExpectedPatch sounds generic, but the patch changes something very specific. Please consider renaming the function to be more specific.

**Comment by fossedihelm** [comment]
> @fossedihelm Thanks for the heads up! shouldn't the test fail in case the code will try to use the custom API calls that aren't implemented by the current client set?

Currently, the generated client will not return any error in such cases.
I will add a panic in  https://github.com/kubevirt/kubevirt/pull/11928 as suggested in previous PR https://github.com/kubevirt/kubevirt/pull/11928.
If you agree I will wait for it to be merged first.

**Comment by orelmisan** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/ca710a598a44bb4305a940ef24ded383fe7130e9..edd8ed096bd006bed5a7ca2c5fae0972cba41d96): Address @RamLavi's [comment](https://github.com/kubevirt/kubevirt/pull/11897#discussion_r1607912646).

File: pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_test.go (hunk 1 above)
Comment (orelmisan) [reply pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_test.go]: Thank you for catching it.
I agree with you.
Done.

**Pull request merged**